### PR TITLE
add try/except block to redshift delete

### PR DESCRIPTION
**Pull request by ocampocj**

### CODE REVIEW

File: c7n/resources/redshift.py [hunk 1]
```diff
@@ -141,7 +141,12 @@ def process_db_set(self, db_set):
             else:
                 params['FinalClusterSnapshotIdentifier'] = snapshot_identifier(
                     'Final', db['ClusterIdentifier'])
-            c.delete_cluster(**params)
+            try:
+                c.delete_cluster(**params)
+            except ClientError as e:
+                if e.response['Error']['Code'] == "InvalidClusterState":
+                    continue
```
Comment (kapilt) [inline c7n/resources/redshift.py]: log these please, warning level

Comment (ocampocj) [inline c7n/resources/redshift.py]: done.  Thanks, @kapilt!

File: c7n/resources/redshift.py [hunk 2]
```diff
@@ -141,7 +141,13 @@ def process_db_set(self, db_set):
             else:
                 params['FinalClusterSnapshotIdentifier'] = snapshot_identifier(
                     'Final', db['ClusterIdentifier'])
-            c.delete_cluster(**params)
+            try:
+                c.delete_cluster(**params)
+            except ClientError as e:
+                if e.response['Error']['Code'] == "InvalidClusterState":
+                    self.log.warning("Exception deleting cluster: %s" %e.response['Error']['Code'])
```
Comment (ocampocj) [inline c7n/resources/redshift.py]: after looking again, I updated the wording to immediately indicate the issue, and I added the clusteridentifier as well.

**Comment by kapilt** [comment]
thanks, lgtm

**Pull request merged**